packages/web3-core/types/index.d.ts

```diff
@@ -0,0 +1,604 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @author Prince Sinha <sinhaprince013@gmail.com>
+ * @date 2018
+ */
+
+import * as net from 'net';
+import { AbstractMethod, AbstractMethodFactory } from 'web3-core-method';
+import BN = require('bn.js');
+
+export class AbstractWeb3Module {
+    constructor(
+        provider: provider,
+        options?: Web3ModuleOptions,
+        methodFactory?: AbstractMethodFactory,
+        net?: net.Socket | null
+    );
+
+    BatchRequest: new () => BatchRequest;
+    defaultBlock: string | number;
+    transactionBlockTimeout: number;
+    transactionConfirmationBlocks: number;
+    transactionPollingTimeout: number;
+    defaultGasPrice: string;
+    defaultGas: number;
+    static readonly providers: Providers;
+    defaultAccount: string | null;
+    readonly currentProvider:
+        | Web3EthereumProvider
+        | HttpProvider
+        | IpcProvider
+        | WebsocketProvider
+        | CustomProvider;
+    readonly givenProvider: any;
+
+    setProvider(provider: provider, net?: net.Socket): boolean;
+
+    isSameProvider(provider: provider): boolean;
+
+    clearSubscriptions(subscriptionType: string): Promise<boolean>;
+}
+
+export interface TransactionSigner {
+    sign(transactionConfig: TransactionConfig): Promise<SignedTransaction>;
+}
+
+export interface SignedTransaction {
+    messageHash?: string;
+    r: string;
+    s: string;
+    v: string;
+    rawTransaction?: string;
+    transactionHash?: string;
+}
+
+export interface Web3ModuleOptions {
+    defaultAccount?: string;
+    defaultBlock?: string | number;
+    transactionBlockTimeout?: number;
+    transactionConfirmationBlocks?: number;
+    transactionPollingTimeout?: number;
+    defaultGasPrice?: string;
+    defaultGas?: number;
+    transactionSigner?: TransactionSigner;
```

Comment:
the defaultGasPrice, defaultGas, and transactionSigner properties do not exist in 1.0